server/block/ice.go

```diff
@@ -0,0 +1,73 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/block/model"
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"math/rand"
+)
+
+// Ice is a transparent block that forms when water freezes and melts when it is near a bright light source.
+type Ice struct {
+	solid
+}
+
+// LightDiffusionLevel ...
+func (Ice) LightDiffusionLevel() uint8 {
+	return 3
+}
+
+// BreakInfo ...
+func (i Ice) BreakInfo() BreakInfo {
+	b := newBreakInfo(0.5, alwaysHarvestable, pickaxeEffective, silkTouchOnlyDrop(i))
+	b.withBreakHandler(func(pos cube.Pos, w *world.World, u item.User) {
```

Comment:
The returned BreakInfo does not have the break handler.

Reply:
My bad

Reply:
Should probably just inline this with the `return`